core/src/Kokkos_BitManipulation.hpp

```diff
@@ -25,69 +25,296 @@
 
 namespace Kokkos::Impl {
 
-template <class T>
-KOKKOS_FUNCTION constexpr T byteswap_fallback(T x) {
-  if constexpr (sizeof(T) > 1) {
-    using U = std::make_unsigned_t<T>;
+#ifndef KOKKOS_COMPILER_NVCC
+#if (defined(KOKKOS_COMPILER_GNU) && (KOKKOS_COMPILER_GNU > 1200)) || \
+    (!defined(KOKKOS_COMPILER_INTEL_LLVM) && defined(__clang__) &&    \
+     (__clang_major__ >= 17))
+#define KOKKOS_IMPL_IF_CONSTEVAL_CXX23_EXTENSION
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wc++23-extensions"
+#elif (defined(KOKKOS_COMPILER_INTEL_LLVM) &&       \
+       (KOKKOS_COMPILER_INTEL_LLVM >= 20240000)) || \
+    (!defined(KOKKOS_COMPILER_INTEL_LLVM) && defined(__clang__))
+#pragma GCC diagnostic push
+#define KOKKOS_IMPL_IF_CONSTEVAL_CXX23_EXTENSION
+#pragma GCC diagnostic ignored "-Wc++2b-extensions"
+#endif
+#endif
```

Comment:
One thing to discuss is whether or not we want to leverage the extension when available.

Reply:
No

Reply:
I thought it would be nice to leverage whenever available but you're the second person commenting about it so I withdrew it.